godel_process_planning/src/common_utils.cpp

```diff
@@ -0,0 +1,257 @@
+  robot_state::RobotState goal_state(model);
+  goal_state.setJointGroupPositions(group, joints_stop);
+
+  moveit_msgs::GetMotionPlan::Request req;
+  req.motion_plan_request.group_name = group_name;
+  req.motion_plan_request.num_planning_attempts = DEFAULT_MOVEIT_NUM_PLANNING_ATTEMPTS;
+  req.motion_plan_request.max_velocity_scaling_factor = DEFAULT_MOVEIT_VELOCITY_SCALING;
+  req.motion_plan_request.allowed_planning_time = DEFAULT_MOVEIT_PLANNING_TIME;
+  req.motion_plan_request.planner_id = DEFAULT_MOVEIT_PLANNER_ID;
+
+  req.motion_plan_request.workspace_parameters.header.frame_id = DEFAULT_MOVEIT_FRAME_ID;
+  req.motion_plan_request.workspace_parameters.header.stamp = ros::Time::now();
+
+  // Set the start state
+  // Will want to add options here to start from a state that's not the start state
+  moveit_msgs::RobotState start_state;
+  start_state.is_diff = false;
+  sensor_msgs::JointState joint_state;
+  joint_state.name = group->getActiveJointModelNames();
+  joint_state.position = joints_start;
+  start_state.joint_state = joint_state;
+  req.motion_plan_request.start_state = start_state;
+
+  // Set the goal state
+  moveit_msgs::Constraints c = kinematic_constraints::constructGoalConstraints(goal_state, group);
+  req.motion_plan_request.goal_constraints.push_back(c);
+
+  // Make connection the planning-service offered by the MoveIt MoveGroup node
+  ros::NodeHandle nh;
+  ros::ServiceClient client = nh.serviceClient<moveit_msgs::GetMotionPlan>(DEFAULT_MOVEIT_PLANNING_SERVICE_NAME);
+
+  trajectory_msgs::JointTrajectory jt;
+  moveit_msgs::GetMotionPlan::Response res;
+  if (client.call(req, res))
+  {
+    jt = res.motion_plan_response.trajectory.joint_trajectory;
+  }
+  else
+  {
+    ROS_WARN("%s: Unable to call MoveIt path planning service: '%s' or planning failed", __FUNCTION__, DEFAULT_MOVEIT_PLANNING_SERVICE_NAME.c_str());
+    throw std::runtime_error("Unable to generate MoveIt path plan");
+  }
+  return jt;
+}
+
+trajectory_msgs::JointTrajectory
+godel_process_planning::planFreeMove(descartes_core::RobotModel& model,
+                                     const std::string& group_name,
+                                     moveit::core::RobotModelConstPtr moveit_model,
+                                     const std::vector<double>& start,
+                                     const std::vector<double>& stop)
+{
+  // Using a mutable model, turns collision checking on for just the
+  // period of this function. Functions called in this function may
+  // throw exceptions and this makes sure the system state is always
+  // valid
+  struct CollisionsGuard {
+    CollisionsGuard(descartes_core::RobotModel& model) : model_(model) {
+      model.setCheckCollisions(true);
+      ROS_WARN_STREAM("Enabling collision");
+    }
+    ~CollisionsGuard() {
+      model_.setCheckCollisions(false);
+      ROS_WARN_STREAM("Disable collision");
+    }
+    descartes_core::RobotModel& model_;
+  };
```

Comment:
This is fancy, but would it not be easier to just enable/disable collision checking directly through the RobotModel method?